web/src/main/java/se/inera/intyg/webcert/web/service/facade/user/UserServiceImpl.java

```diff
@@ -61,50 +61,96 @@ public User getLoggedInUser() {
         final var loggedInCareProvider = getLoggedInCareProvider(webCertUser);
         final var params = webCertUser.getParameters();
         final var isInactiveUnit = params != null && params.isInactiveUnit();
-
-        return User.builder()
-            .hsaId(webCertUser.getHsaId())
-            .name(webCertUser.getNamn())
-            .role(getRole(webCertUser))
-            .loggedInUnit(
-                Unit.builder()
-                    .unitName(
-                        loggedInUnit.getNamn()
-                    )
-                    .unitId(
-                        loggedInUnit.getId()
-                    )
-                    .isInactive(
-                        isInactiveUnit
-                    )
-                    .build()
-            )
-            .loggedInCareUnit(
-                Unit.builder()
-                    .unitName(
-                        loggedInCareUnit.getNamn()
-                    )
-                    .unitId(
-                        loggedInCareUnit.getId()
-                    )
-                    .build()
-            )
-            .loggedInCareProvider(
-                Unit.builder()
-                    .unitName(
-                        loggedInCareProvider.getNamn()
-                    )
-                    .unitId(
-                        loggedInCareProvider.getId()
-                    )
-                    .build()
-            )
-            .protectedPerson(webCertUser.isSekretessMarkerad())
-            .preferences(webCertUser.getAnvandarPreference())
-            .loginMethod(getLoginMethod(webCertUser.getAuthenticationMethod()))
-            .signingMethod(getSigningMethod(webCertUser.getAuthenticationMethod()))
-            .careProviders(getCareProviders(webCertUser))
-            .build();
+        if (params != null && params.getLaunchId() != null) {
```

Comment:
Try restructure the code so you don't have to create this if-else statement. For example, see how "isInactiveUnit" has been implemented.